scvi/harmonization/utils.py

```diff
@@ -0,0 +1,94 @@
+from numpy import loadtxt
+from scipy.io import mmread
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from scvi.dataset.dataset import GeneExpressionDataset
+import igraph as ig
+import louvain
+from sklearn.neighbors import kneighbors_graph
+
+
+def get_matrix_from_dir(dirname, storage='../'):
+    geneid = loadtxt(storage + dirname + '/genes.tsv', dtype='str', delimiter="\t")
+    cellid = loadtxt(storage + dirname + '/barcodes.tsv', dtype='str', delimiter="\t")
+    count = mmread(storage + dirname + '/matrix.mtx')
+    return count, geneid, cellid
+
+
+class ExpressionMatrix(GeneExpressionDataset):
+    def __init__(self, filename, save_path):
+        self.save_path = save_path
+        self.filename = filename
+        count, barcode, genenames = self.preprocess()
+        super(ExpressionMatrix, self).__init__(
```

Comment:
Why do we need these objects ?

Reply:
Should this be in the `dataset` directory? (Or can we delete it entirely?)